From 4c27799d7f2c8b1a5d991a33fd1cc7f159668117 Mon Sep 17 00:00:00 2001
From: Piotr Gorski <lucjan.lucjanov@gmail.com>
Date: Sun, 22 Mar 2020 20:58:03 +0100
Subject: [PATCH 6/6] Revert "block, bfq: fix overwrite of bfq_group pointer in
 bfq_find_set_group()"

This reverts commit cc9f2879f3bbbc9007a8f0e03ccf2b6e03c57229.

Signed-off-by: Piotr Gorski <lucjan.lucjanov@gmail.com>
---
 block/bfq-cgroup.c | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/block/bfq-cgroup.c b/block/bfq-cgroup.c
index 1aef1af08137..e1419edde2ec 100644
--- a/block/bfq-cgroup.c
+++ b/block/bfq-cgroup.c
@@ -610,13 +610,12 @@ struct bfq_group *bfq_find_set_group(struct bfq_data *bfqd,
 	 */
 	entity = &bfqg->entity;
 	for_each_entity(entity) {
-		struct bfq_group *curr_bfqg = container_of(entity,
-						struct bfq_group, entity);
-		if (curr_bfqg != bfqd->root_group) {
-			parent = bfqg_parent(curr_bfqg);
+		bfqg = container_of(entity, struct bfq_group, entity);
+		if (bfqg != bfqd->root_group) {
+			parent = bfqg_parent(bfqg);
 			if (!parent)
 				parent = bfqd->root_group;
-			bfq_group_set_parent(curr_bfqg, parent);
+			bfq_group_set_parent(bfqg, parent);
 		}
 	}
 
-- 
2.26.0.rc2.dirty

